### PR TITLE
Improved /settingslist + fixed /settings pt.1

### PR DESCRIPTION
**Pull request by mctaylors**
This PR is mainly aimed at improving /settingslist and fixing /settings

List of things to do before merging:
- [x] #62
- [x] Add the pages feature to /settingslist
- [x] Add bullets like these -> 

And since the development has already been taking more than 2 days, I suggest splitting the PR into 2 parts.

List of other things that will be in the future PR:
- mctaylors#1
- Fix bot not answering when an invalid setting is specified in /settings
- Options list for /settings

### CODE REVIEW

**Review by neroduckale — APPROVED** [review]

File: src/Commands/SettingsCommandGroup.cs [hunk 1]
```diff
@@ -80,23 +81,40 @@ public class SettingsCommandGroup : CommandGroup {
...
+        const int optionsPerList = 7;
+        var totalPages = (AllOptions.Length + optionsPerList - 1)/optionsPerList;
+        for (var i = optionsPerList * page - optionsPerList; i <= optionsPerList * page - 1; i++) {
+            try {
+                builder.AppendLine($"Settings{AllOptions[i].Name}".Localized())
+                    .Append(Markdown.InlineCode(AllOptions[i].Name))
+                    .Append(": ")
+                    .AppendLine(AllOptions[i].Display(cfg))
+                    .AppendLine();
+            } catch { /* hilariously ignored */ }
```
Comment (Octol1ttle) [inline src/Commands/SettingsCommandGroup.cs:99]: I do not like this presence of try-catch without any explanation while also completely ignoring any exception that occurs.

File: src/Commands/SettingsCommandGroup.cs [hunk 2]
```diff
@@ -80,23 +81,40 @@ public class SettingsCommandGroup : CommandGroup {
...
-            builder.AppendLine(option.Display(cfg));
+        var footer = new StringBuilder();
+        const int optionsPerList = 7;
+        var totalPages = (AllOptions.Length + optionsPerList - 1)/optionsPerList;
+        for (var i = optionsPerList * page - optionsPerList; i <= optionsPerList * page - 1; i++) {
+            try {
+                builder.AppendLine($"Settings{AllOptions[i].Name}".Localized())
+                    .Append(Markdown.InlineCode(AllOptions[i].Name))
+                    .Append(": ")
+                    .AppendLine(AllOptions[i].Display(cfg))
```
Comment (Octol1ttle) [inline src/Commands/SettingsCommandGroup.cs:97]: This is a lot of array accesses that should be extracted in a variable

File: locale/Messages.resx [hunk 3]
```diff
@@ -561,4 +561,19 @@
     <data name="SettingsRenameHoistedUsers" xml:space="preserve">
         <value>Rename members who attempt to hoist themselves</value>
     </data>
+    <data name="Page" xml:space="preserve">
+        <value>page</value>
```
Comment (Octol1ttle) [inline locale/Messages.resx:565]: "Page"

File: locale/Messages.resx [hunk 4]
```diff
@@ -561,4 +561,19 @@
...
+    </data>
+    <data name="PageNotFound" xml:space="preserve">
+        <value>Page not found!</value>
+    </data>
+    <data name="PagesAllowed" xml:space="preserve">
+        <value>There are total pages</value>
```
Comment (Octol1ttle) [inline locale/Messages.resx:571]: "There are {0} total pages"

File: locale/Messages.ru.resx [hunk 5]
```diff
@@ -561,4 +561,19 @@
...
         <value>Переименовывать участников, которые пытаются поднять себя</value>
     </data>
+    <data name="Page" xml:space="preserve">
+        <value>страница</value>
+    </data>
+    <data name="PageNotFound" xml:space="preserve">
+        <value>Страница не найдена!</value>
+    </data>
+    <data name="PagesAllowed" xml:space="preserve">
+        <value>Всего страниц существует</value>
```
Comment (Octol1ttle) [inline locale/Messages.ru.resx:571]: "Всего есть {0} страниц"

File: locale/Messages.tt-ru.resx [hunk 6]
```diff
@@ -561,4 +561,19 @@
...
         <value>переобувать шизоидов пытающихся поднять себя в табе</value>
     </data>
+    <data name="Page" xml:space="preserve">
+        <value>это страница</value>
+    </data>
+    <data name="PageNotFound" xml:space="preserve">
+        <value>если я был бы html, я бы сказал 404</value>
+    </data>
+    <data name="PagesAllowed" xml:space="preserve">
+        <value>ну а если быть точнее, тут всего страниц</value>
```
Comment (Octol1ttle) [inline locale/Messages.tt-ru.resx:571]: "ну а если быть точнее, тут всего {0} страниц"

File: src/Commands/SettingsCommandGroup.cs [hunk 7]
```diff
@@ -80,23 +81,40 @@ public class SettingsCommandGroup : CommandGroup {
...
+    private async Task<Result> SendSettingsListAsync(JsonNode cfg, IUser currentUser, int page, CancellationToken ct = default) {
         var builder = new StringBuilder();
-
-        foreach (var option in AllOptions) {
-            builder.Append(Markdown.InlineCode(option.Name))
-                .Append(": ");
-            builder.AppendLine(option.Display(cfg));
+        var footer = new StringBuilder();
+        const int optionsPerList = 7;
+        var totalPages = (AllOptions.Length + optionsPerList - 1)/optionsPerList;
```
Comment (Octol1ttle) [inline src/Commands/SettingsCommandGroup.cs:91]: Missing spacing between operators

File: locale/Messages.ru.resx [hunk 8]
```diff
@@ -561,4 +561,19 @@
     <data name="SettingsRenameHoistedUsers" xml:space="preserve">
         <value>Переименовывать участников, которые пытаются поднять себя</value>
     </data>
+    <data name="Page" xml:space="preserve">
+        <value>страница</value>
```
Comment (Octol1ttle) [inline locale/Messages.ru.resx:565]: "Страница"

File: src/Commands/SettingsCommandGroup.cs [hunk 9]
```diff
@@ -80,23 +81,40 @@ public class SettingsCommandGroup : CommandGroup {
...
+        for (var i = optionsPerList * page - optionsPerList; i <= optionsPerList * page - 1; i++) {
```
Comment (Octol1ttle) [inline src/Commands/SettingsCommandGroup.cs:92]: Extract the result of these calculations into separate variables for more clarity

File: src/Commands/SettingsCommandGroup.cs [hunk 10]
```diff
@@ -80,23 +81,40 @@ public class SettingsCommandGroup : CommandGroup {
...
             .WithColour(ColorsList.Default)
+            .WithFooter(footer.ToString())
             .Build();
 
+        if (optionsPerList * page - optionsPerList >= AllOptions.Length) {
+            embed = new EmbedBuilder().WithSmallTitle(Messages.PageNotFound, currentUser)
+                .WithDescription($"{Messages.PagesAllowed}: {Markdown.Bold(totalPages.ToString())}")
+                .WithColour(ColorsList.Red)
+                .Build();
+        }
```
Comment (Octol1ttle) [inline src/Commands/SettingsCommandGroup.cs:116]: Why is this being checked so late?

File: src/Commands/SettingsCommandGroup.cs [hunk 11]
```diff
@@ -80,23 +81,40 @@ public class SettingsCommandGroup : CommandGroup {
...
+        var footer = new StringBuilder();
+        const int optionsPerList = 7;
+        var totalPages = (AllOptions.Length + optionsPerList - 1)/optionsPerList;
+        for (var i = optionsPerList * page - optionsPerList; i <= optionsPerList * page - 1; i++) {
+            try {
+                builder.AppendLine($"Settings{AllOptions[i].Name}".Localized())
+                    .Append(Markdown.InlineCode(AllOptions[i].Name))
+                    .Append(": ")
+                    .AppendLine(AllOptions[i].Display(cfg))
+                    .AppendLine();
```
Comment (Octol1ttle) [inline src/Commands/SettingsCommandGroup.cs:98]: A lot of space is being wasted here due to excess newlines. Consider implementing a design similar to pre-Remora Boyfriend.

**Pull request merged**